src/http/ngx_http_mruby_module.c

```diff
@@ -1889,6 +1891,89 @@ static ngx_int_t ngx_http_mruby_read_body(ngx_http_request_t *r, ngx_chain_t *in
 
 #if (NGX_HTTP_SSL) && OPENSSL_VERSION_NUMBER >= 0x1000205fL
 
+static int ngx_http_mruby_set_der_certificate_data(ngx_ssl_conn_t *ssl_conn, ngx_str_t *cert, ngx_str_t *key)
+{
+  BIO *bio = NULL;
+  EVP_PKEY *pkey = NULL;
+  X509 *x509 = NULL;
+  u_long n;
+
+  /* read certficate data from memory buffer */
+  if ((bio = BIO_new_mem_buf(cert->data, cert->len)) == NULL) {
+    BIO_free(bio);
+    return NGX_ERROR;
+  }
+
+  if ((x509 = PEM_read_bio_X509_AUX(bio, NULL, NULL, NULL)) == NULL) {
+    BIO_free(bio);
+    return NGX_ERROR;
+  }
+
+  SSL_certs_clear(ssl_conn);
+
+  if (SSL_use_certificate(ssl_conn, x509) == 0) {
+    X509_free(x509);
+    BIO_free(bio);
+    return NGX_ERROR;
+  }
+
+  X509_free(x509);
+  x509 = NULL;
+
+  /* read rest of the chain */
+  for (;;) {
+    x509 = PEM_read_bio_X509(bio, NULL, NULL, NULL);
```

Comment:
I understood your point. So, I add more error handling to avoid memory leak as possible via https://github.com/matsumoto-r/ngx_mruby/commit/603459ba828658c375c9325eba0a43bc9fa94141 .

Thank you for your notice.